x/distribution/abci_app.go

```diff
@@ -0,0 +1,25 @@
+package distribution
+
+import (
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+)
+
+// set the proposer for determining distribution during endblock
+func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper) {
+	consAddr := sdk.ConsAddress(req.Header.ProposerAddress)
+	k.SetProposerConsAddr(ctx, consAddr)
+
+	// XXX TODO actually calculate this
```

Comment:
Let's pass it through from slashing maybe? We iterate through precommits there anyways.

Reply:
dup comment